src/Raven.Server/Documents/Indexes/Persistence/IndexReadOperationBase.cs

```diff
@@ -70,8 +71,14 @@ public override void Dispose()
                 var mangedDiff = GC.GetAllocatedBytesForCurrentThread() - _memoryInfo.AllocatedManagedBefore;
                 var unmanagedDiff = Math.Max(0, NativeMemory.ThreadAllocations.Value.TotalAllocated - _memoryInfo.AllocatedUnmanagedBefore);
 
+                if (_logger.IsOperationsEnabled && mangedDiff < ThresholdForMemoryUsageLoggingInBytes && unmanagedDiff < ThresholdForMemoryUsageLoggingInBytes)
```

Comment:
Why is this check only on `IsOperlationsEnabled` ?

Reply:
check prev if statement